src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs

```diff
@@ -14,54 +14,501 @@
 
 using System.Collections;
 using System.Management.Automation;
-using System.Security.Permissions;
 using Microsoft.Azure.Commands.DataFactoryV2.Models;
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.DataFactory.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
 
 namespace Microsoft.Azure.Commands.DataFactoryV2
 {
-    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataFactoryV2", SupportsShouldProcess = true), OutputType(typeof(PSDataFactory))]
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataFactoryV2", DefaultParameterSetName = ParameterSetNames.ByFactoryName, SupportsShouldProcess = true), OutputType(typeof(PSDataFactory))]
     [Alias("New-" + ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataFactoryV2")]
     public class SetAzureDataFactoryCommand : DataFactoryBaseCmdlet
     {
-        [Parameter(Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true,
+        #region Attributes
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceId,
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpResourceId)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceIdFactoryRepoVstsConfig,
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpResourceId)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceIdFactoryRepoGitConfig,
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpResourceId)]
+        [ValidateNotNullOrEmpty]
+        [Alias(Constants.Id, Constants.DataFactoryId)]
+        #endregion // Attributes
+        public string ResourceId { get; set; }
```

Comment:
You may want to add [ResourceIdCompleter("<resource_type>")] attribute to the parameter.

Reply:
Done.